python/ray/_private/accelerators/amd_gpu.py

```diff
@@ -54,32 +53,35 @@ def get_current_process_visible_accelerator_ids() -> Optional[List[str]]:
 
     @staticmethod
     def get_current_node_num_accelerators() -> int:
+        import ray._private.thirdparty.pyamdsmi as pyamdsmi
+
         num_gpus = 0
 
-        if sys.platform.startswith("linux"):
+        try:
+            pyamdsmi.smi_initialize()
```

Comment:
What error will it raise if this code runs on non-amd GPU machines?

Reply:
smi_initialize() will throw either RuntimeError('ROCm SMI initialization failed') or RuntimeError('ROCm driver initilization failed')